### PR TITLE
Feature/javabinding testframework add platforms

### PR DESCRIPTION
**Pull request by euniceatcouchbase**
#### Phase I of Java Binding test automation support

- [X] Ran `flake8`
- [X] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- this PR is to add java and java webservice platform support
- implementation for download/install/start java/java-ws apps will be in a separate PR.
- added java binding support in test suites where special platform occasion need to deal with.

### CODE REVIEW

File: keywords/TestServerNetMsft.py [hunk 1]
```diff
@@ -1,16 +1,16 @@
 import os
 import time
 
-from keywords.TestServerBase import TestServerBase
+from keywords.TestServerWinBase import TestServerWinBase
 from keywords.constants import LATEST_BUILDS, RELEASED_BUILDS
 from keywords.exceptions import LiteServError
-from keywords.utils import version_and_build
+# from keywords.utils import version_and_build
```
Comment (mgcouchbase) [inline keywords/TestServerNetMsft.py]: delete commented imports

File: testsuites/CBLTester/CBL_Functional_tests/conftest.py [hunk 2]
```diff
@@ -675,7 +675,7 @@ def params_from_base_test_setup(request, params_from_base_suite_setup):
     if request.node.rep_call.failed and enable_file_logging and create_db_per_test is not None:
         test_id = request.node.nodeid
         log_info("\n Collecting logs for failed test: {}".format(test_id))
-        zip_data = test_cbllog.get_logs_in_zip(test_db_log_file)
+        zip_data = test_cbllog.get_logs_in_zip()
```
Comment (mgcouchbase) [inline testsuites/CBLTester/CBL_Functional_tests/conftest.py]: is it required ?

Comment (euniceatcouchbase) [reply testsuites/CBLTester/CBL_Functional_tests/conftest.py]: no parameter is required to pass, confirmed with Hemant

File: keywords/TestServerNetMsft.py (hunk 1 above)
Comment (euniceatcouchbase) [reply keywords/TestServerNetMsft.py]: done

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_database.py [hunk 3]
```diff
@@ -328,6 +328,10 @@ def test_copy_prebuilt_database(params_from_base_test_setup, encrypted):
         prebuilt_db_path = "/assets/{}.cblite2.zip".format(db_prefix)
     elif liteserv_platform == "xamarin-android":
         prebuilt_db_path = "{}.cblite2.zip".format(db_prefix)
+    elif liteserv_platform in ["java-macosx", "java-msft", "java-ubuntu", "java-centos"]:
+        prebuilt_db_path = "{}.cblite2.zip".format(db_prefix)
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_database.py]: line #332 and #334 looks same as line 330. you may need to rearrange , you can put "net-msft" condition and "ios" condition before and rest all in else condition with the logic which is in in line #332

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_delta_sync.py [hunk 4]
```diff
@@ -107,6 +107,9 @@ def test_delta_sync_replication(params_from_base_test_setup, num_of_docs, replic
                     if liteserv_platform == "android":
                         image_content = blob.createImageContent("/assets/golden_gate_large.jpg")
                         blob_value = blob.create("image/jpeg", stream=image_content)
+                    elif liteserv_platform in ["java-macosx", "java-msft", "java-ubuntu", "java-centos", "javaws-macosx", "javaws-msft", "javaws-ubuntu", "javaws-centos"]:
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_delta_sync.py]: this logic is same as "xamarin-android". Can you combine them?

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py [hunk 5]
```diff
@@ -3120,11 +3126,9 @@ def test_resetCheckpointFailure(params_from_base_test_setup):
     # verify it throws an error that checkpoint reset is called without stopping replicator.
     with pytest.raises(Exception) as he:
         replicator.resetCheckPoint(repl)
-    if liteserv_platform.lower() == "android":
-        assert 'Attempt to reset the checkpoint for a replicator that is not stopped.' in he.value.message, "Reset the checkpoint should have thrown exception to inform that replicator is not stopped."
-    else:
-        assert 'Replicator is not stopped.' in he.value.message
-        assert 'Resetting checkpoint is only allowed when the replicator is in the stopped state' in he.value.message, "Reset the checkpoint should have thrown exception to inform that replicator is not stopped."
+
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py]: "android" condition is removed, In that case, wouldn't this assertion fail when it runs against "android"?

File: testsuites/CBLTester/System_test_multiple_device/conftest.py [hunk 6]
```diff
@@ -19,6 +19,11 @@
 
 
 def pytest_addoption(parser):
+    parser.addoption("--use-local-testserver",
```
Comment (sridevi-15) [inline testsuites/CBLTester/System_test_multiple_device/conftest.py]: Thanks for adding this

File: testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py [hunk 7]
```diff
@@ -341,6 +341,8 @@ def _upgrade_db(args):
         prebuilt_db_path = "/assets/{}.cblite2.zip".format(old_liteserv_db_name)
     elif liteserv_platform == "xamarin-android":
         prebuilt_db_path = "{}.cblite2.zip".format(old_liteserv_db_name)
+    elif liteserv_platform in ["java-macosx", "java-msft", "java-ubuntu", "java-centos", "javaws-macosx", "javaws-msft", "javaws-ubuntu", "javaws-centos"]:
```
Comment (sridevi-15) [inline testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py]: same logic as 'xamarin-android'. can you combine or else you can rearrange and have this on else condition

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py (hunk 5 above)
Comment (euniceatcouchbase) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py]: android used deal error message in its own format, javabinding code refactor made the error output same as other platforms. now if condition is needed then.

**Pull request merged**